### PR TITLE
Omniauth sso for user log on

### PR DESCRIPTION
**Pull request by Allegraknox**
-Added omniauth gem
-Added google authentication (will need to do extra configuration steps for our next Heroku deploy)
-Only @colgate.edu emails will be allowed to log-in
-No access for non-sso/existing accounts (i.e. Brian casey: email@email.com cannot log in anymore but we can still use him for db seeds be listed as a "seller" or whatever)
-User can edit their profile information (only name and payment prefs--not email)
-Changed "log in/log out/edit profile" buttons from index view to application layouts view
-Changed route for edit/update user page (using usersController instead of built in devise)
-If user has not yet input their name, "you are logged in as XXX" displays email instead of name

-NOTE: items_controller added a few lines because newly created items were not being shown in index view: the issue was that "listed" was not being saved in params, so it was never being shown (even though it was created correctly).

### CODE REVIEW

File: app/views/users/registrations/edit.html.erb [hunk 1]
```diff
@@ -50,4 +50,4 @@
 
 <p>Unhappy? <%= button_to "Delete my account", registration_path(resource_name), method: :delete %></p>
 
-<%= link_to "Back", :back %>
+<%= link_to "Back", root_path %>
```
Comment (Allegraknox) [inline app/views/users/registrations/edit.html.erb:53]: User doesn't have access to this view anymore so can ignore this (using custom edit view now not the devise view)

File: app/views/users/registrations/new.html.erb [hunk 2]
```diff
@@ -36,4 +36,4 @@
   </div>
 <% end %>
 
-<%= render "users/shared/links" %>
+<%= render "users/shared/links" %>
```
Comment (Allegraknox) [inline app/views/users/registrations/new.html.erb:39]: Same comment as with registrations/edit.html.erb

File: app/controllers/items_controller.rb [hunk 3]
```diff
@@ -28,8 +28,10 @@ def new
   
   def create
     params[:item][:status] = "available"
+    params[:item][:listed] = true
```
Comment (amycsong98) [inline app/controllers/items_controller.rb:31]: nice catch!

**Review by amycsong98 — APPROVED** [review]
looks really good!

**Pull request merged**